src/Controls/tests/Core.UnitTests/RadioButtonTests.cs

```diff
@@ -236,5 +243,43 @@ public void GroupSelectedValueUpdatesWhenSelectedButtonValueUpdates()
 
 			Assert.Equal("updated", layout.GetValue(RadioButtonGroup.SelectedValueProperty));
 		}
+
+		[Fact]
+		public async Task RadioButtonGroupShouldNotLeakWhenRemoved()
+		{
+			WeakReference CreateReference()
+			{
+				var layout = new StackLayout();
+				var rb1 = new RadioButton { Content = "RB1", Value = "RB1" };
+				var rb2 = new RadioButton { Content = "RB2", Value = "RB2" };
+				layout.Add(rb1);
+				layout.Add(rb2);
+				layout.SetValue(RadioButtonGroup.GroupNameProperty, "GroupA");
+				layout.SetValue(RadioButtonGroup.SelectedValueProperty, "RB1");
+
+				return new(layout);
+			}
+
+			WeakReference reference = CreateReference();
+
+			await TestHelpers.Collect();
+
+			Assert.False(reference.IsAlive, "RadioButtonGroup should not be alive");
+		}
+
+		[Fact]
+		public async Task RadioButtonShouldNotLeak()
+		{
+			WeakReference CreateReference()
+			{
+				var radioButton = new RadioButton { Content = "RB1", Value = "RB1" };
+				return new(radioButton);
```

Comment:
Can you set `RadioButtonGroup.GroupName` on this `RadioButton`?

Reply:
> Can you set `RadioButtonGroup.GroupName` on this `RadioButton`?

I had already set the `RadioButtonGroup.GroupName` on the parent container in a previous test. In this test, I am now setting the `RadioButton.GroupName` property directly.